WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorViewController.swift

```diff
@@ -0,0 +1,127 @@
+    private var dataSource: DataSource
+    private let onDismiss: (_ selected: Model?) -> Void
+
+    private let rowType = Cell.self
+
+    private let estimatedSectionHeight = CGFloat(44)
+
+    @IBOutlet private(set) weak var tableView: UITableView!
+
+    init(viewProperties: BottomSheetListSelectorViewProperties,
+         dataSource: DataSource,
+         onDismiss: @escaping (_ selected: Model?) -> Void) {
+        self.viewProperties = viewProperties
+        self.dataSource = dataSource
+        self.onDismiss = onDismiss
+        super.init(nibName: "BottomSheetListSelectorViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureMainView()
+        configureTableView()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        onDismiss(dataSource.selected)
+        super.viewWillDisappear(animated)
+    }
+
+    // MARK: UITableViewDataSource
+    //
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return dataSource.data.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: rowType.reuseIdentifier,
+                                                       for: indexPath) as? Cell else {
+                                                        fatalError()
+        }
+        let model = dataSource.data[indexPath.row]
+        dataSource.configureCell(cell: cell, model: model)
+
+        return cell
+    }
+
+    // MARK: UITableViewDelegate
+    //
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let selected = dataSource.data[indexPath.row]
+        if selected != dataSource.selected {
+            dataSource.handleSelectedChange(selected: selected)
+            tableView.reloadData()
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: BottomSheetListSelectorSectionHeaderView.reuseIdentifier)
+            as? BottomSheetListSelectorSectionHeaderView else {
+                fatalError()
+        }
+        header.configure(text: viewProperties.title)
+        return header
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return estimatedSectionHeight
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
```

Comment:
Since these are not dynamic, we could probably save a few lines by using these properties in `configureTableView()` instead:

- [estimatedSectionHeaderHeight](https://developer.apple.com/documentation/uikit/uitableview/1614957-estimatedsectionheaderheight)
- [sectionHeaderHeight](https://developer.apple.com/documentation/uikit/uitableview/1614863-sectionheaderheight)

Reply:
yep fewer lines of code! updated in https://github.com/woocommerce/woocommerce-ios/pull/2155/commits/7b1c0dc374880bd2e14eb8ef035be9d5e68d7d1f